pontoon/base/migrations/0001_squashed_0154_auto_20200206_1736.py

```diff
@@ -1200,8 +1200,8 @@
 ]
 
 MIGRATION_0146_USERS = [
-    ("pontoon-tm@mozilla.com", "translation-memory", "Translation Memory"),
-    ("pontoon-gt@mozilla.com", "google-translate", "Google Translate"),
+    ("pontoon-tm@example.com", "translation-memory", "Translation Memory"),
+    ("pontoon-gt@example.com", "google-translate", "Google Translate"),
```

Comment:
Note that changing migration files is prohibited.

Instead you need to add a new migration in the `base` app, in this case a [data migration](https://docs.djangoproject.com/en/3.0/topics/migrations/#data-migrations), that will change the email addresses of both affected users.

In the instructions above see how you create a migration. Please change its name to something semantic, e.g. `0009_change_pontoon_users_emails.py`. Note that the inital 4 characters must remain unchanged - they need to increase by one with each new migration.

Contents of the migration should be very similar to https://github.com/mozilla/pontoon/blob/f15948a3e27cdce6a3dc3cd7dd043b8b523535e9/pontoon/sync/migrations/0002_change_pontoon_sync_email.py, except that we'll be changing different users.

Reply:
Hi, I have created a new data migration to the base app named 0009_change_pontoon_users_emails.py. Before proceeding any further could you please have a look at my code. Is this the correct way to "change different users" as mentioned by you

# Generated by Django 2.2.13 on 2020-08-28 08:49
from __future__ import unicode_literals

from django.db import migrations

from django.db.models import Q


def update_base_user_email(apps, schema_editor):
    User = apps.get_model("auth", "User")
    User.objects.filter(
        (Q(email="pontoon-sync@mozilla.org").update(email="pontoon-sync@example.com") |
        Q(email="pontoon-gt@mozilla.com").update(email="pontoon-gt@example.com") |
        Q(email="pontoon-tm@mozilla.com").update(email="pontoon-tm@example.com")
    )
    )

def revert_base_user_email(apps, schema_editor):
    User = apps.get_model("auth", "User")
    User.objects.filter(
        (Q(email="pontoon-sync@example.com").update(email="pontoon-sync@mozilla.org") |
        Q(email="pontoon-gt@example.com").update(email="pontoon-gt@mozilla.com") |
        Q(email="pontoon-tm@example.com").update(email="pontoon-tm@mozilla.com")
        )
    )

class Migration(migrations.Migration): 

    dependencies = [
        ('base', '0008_add_systran_locales'),
    ]

    operations = [
        migrations.RunPython(
            code=update_base_user_email, reverse_code=revert_base_user_email,
        ),
    ]

Reply:
You need to make separate DB query for each user:
```python
User.objects.filter(email="pontoon-sync@mozilla.com").update(email="pontoon-sync@example.com")
User.objects.filter(email="pontoon-gt@mozilla.com").update(email="pontoon-gt@example.com")
User.objects.filter(email="pontoon-mt@mozilla.com").update(email="pontoon-mt@example.com")
```

Obviously, you'll need similar (but reverse) queries in `revert_base_user_email()`.

Also note that you can [style blocks of code](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#syntax-highlighting) on GitHub as code for better readability.

Reply:
Okay Noted.